modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ConstructMessageProcessor.java

```diff
@@ -124,225 +36,25 @@ public ConstructMessageProcessor(ExtensionModel extensionModel,
                                    RetryPolicyTemplate retryPolicyTemplate,
                                    ExtensionManager extensionManager,
                                    PolicyManager policyManager) {
-    super(extensionModel, constructModel, configurationProvider, cursorProviderFactory, extensionManager);
-
-    this.classLoader = getClassLoader(extensionModel);
-    this.extensionManager = extensionManager;
-    this.extensionModel = extensionModel;
-    this.constructModel = constructModel;
-    this.resolverSet = resolverSet;
-    this.target = target;
-    this.targetValue = targetValue;
-    this.policyManager = policyManager;
-    this.retryPolicyTemplate = retryPolicyTemplate;
-    this.cursorProviderFactory = cursorProviderFactory;
+    super(extensionModel, constructModel, configurationProvider, target, targetValue,
+          resolverSet, cursorProviderFactory, retryPolicyTemplate, extensionManager, policyManager);
   }
 
   @Override
-  public InternalEvent process(InternalEvent event) throws MuleException {
-    return processToApply(event, this);
+  protected ReturnDelegate createReturnDelegate() {
+    return !isTargetPresent() ? getValueReturnDelegate() : getTargetReturnDelegate();
```

Comment:
will constructs ever have a target? This seems to assume no router will ever return void. Seems to me like the implementation of OperationMessageProcessor is also good for this one and should actually be the one in the base class